pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [validation of column ... NOT NULL failed]

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/0eeffa47-23af-4a55-8df5-99a895a54ff5?testReportRun=25&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on release-24.3 @ [3bb549350582192ffa091bfecba15527e545e30b](https://github.com/cockroachdb/cockroach/commits/3bb549350582192ffa091bfecba15527e545e30b):


```
  },
  "COMMIT"
 ],
 "expectedExecErrors": "",
 "expectedCommitErrors": "",
 "message": "TXN RETRY ERROR; ERROR: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh due to encountered recently written committed value /Table/3/1/126/2/1 @1737790503.767205585,0): \"sql txn\" meta={id=44d79d0e key=/Table/3/1/105/2/1 iso=Serializable pri=0.13484689 epo=0 ts=1737790503.802520625,1 min=1737790503.714087831,0 seq=13} lock=true stat=PENDING rts=1737790503.714087831,0 wto=false gul=1737790504.214087831,0 (SQLSTATE 40001)"
}
{
 "workerId": 1,
 "clientTimestamp": "07:35:03.840568",
 "ops": [
  "BEGIN",
  {
   "sql": "SELECT t0.col139_w1_140 AS col0, t0.col139_w1_143 AS col1, t0.col139_w1_141 AS col2, t0.col139_w1_143 AS col3 FROM public.table_w1_139 AS t0 FETCH FIRST 1 ROWS ONLY",
   "potentialExecErr": "53100,53200,57014"
  }
 ],
 "expectedExecErrors": "",
 "expectedCommitErrors": "",
 "message": "ERROR: internal error: requested column col139_w1_141 not in index (SQLSTATE XX000)"
}
    schema_change_external_test.go:131: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:131
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR: ERROR: internal error: requested column col139_w1_141 not in index (SQLSTATE XX000)
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:626
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:120
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) ***UNEXPECTED ERROR
        	            	Wraps: (3) ERROR: internal error: requested column col139_w1_141 not in index (SQLSTATE XX000)
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 73
Total Schema Statements Succeeded = 73
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 44
Total Transactions Rolled Back = 35
Total Transactions Executed = 79
    schema_change_external_test.go:96: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_25_of_25/test.outputs/logTestWorkload1011296832
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1011296832
--- FAIL: TestWorkload (8.92s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=25</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #136966 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [duplicate key value violates unique constraint] [C-test-failure O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46834

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://tanzanite.cluster.engflow.com/invocations/default/0f42ab12-5777-4bd1-86ca-1c809f83efda?testReportRun=5&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on release-24.3 @ [303bfcd84f0820ab42ac5bc2384e93ad90160d57](https://github.com/cockroachdb/cockroach/commits/303bfcd84f0820ab42ac5bc2384e93ad90160d57):


```
 "clientTimestamp": "06:12:05.380038",
 "ops": [
  "BEGIN",
  {
   "sql": "ALTER TABLE schema_w1_73.table_w0_265 ALTER PRIMARY KEY USING COLUMNS (col265_w0_266, col265_w0_272, col265_w0_269) USING HASH",
   "potentialExecErr": "42701,42P10"
  },
  "COMMIT"
 ],
 "expectedExecErrors": "",
 "expectedCommitErrors": "",
 "message": ""
}
    schema_change_external_test.go:131: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:131
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: validation of column "col176_w1_192" NOT NULL failed on row: col47_w0_62='54aa7523-0fb1-4d47-89d1-325341d96435', col47_w0_50='05:41:14.984757', col47_w0_48=!'wVV', col176_w1_184=1266, col176_w1_185='1992-01-21', col176_w1_192=NULL, col176_w1_178=[1.8291579e+25,4.576669e+36,4.027279e+22,0.00012290478,2.76803e-05,1.02607264e+09,1.1482619e+20,2.9291637e+36,-1.832785e-21,605456.94,-7.336126e+34,7.9806176e+12,7.9217315,-7.926753e-38,-4.785446e-15,-4.2700513e+36,-2.1663763e+15,-7.703745e+31,-8.5369656e+09,-0.00011374854,-8.525576e-31,2.196886e-23,1.4807279e-25,1.0925293e+13,-2.1461318e-38,1.5555648e+22,3.0977012e-20,-1.6310505e-35,119088.17,-6.9775663e+25,-1.9890994e+22,-842812.6,1.5902551e+16,-8.4925106e+37,9.262322e-16,-2.146512e-30,1.409276e+37,7.3536126e-37,-4.751217e-27,-6.915994e+13,-3.2004928e+11,7.468967e+13,4.9822395e+33,-6.796619e-10,2.4918778e+08,1.0572669e-18,-11.972408,2.1249514e+20,1.3580067e+31,-4.13354e-39,2.860174e+19,9.7524676e-08,1.281487e-17,5.7726207,-1.9020512e+22,-1.3046642e+22,-1.2268314e+10,2.7600295e-19,22843.742,3.2023303e-07,-844.45245,-2.8693878e+34,-7.3299745e-08,-1.7874349e-34,1.4188787e-36,7.96708e+14,7.305435e+07,1.6642996e-05,7.381059e-31,-7.09088e-07,4.1403404e-29,2.1951314e+09,5.1102694e+08,-4.0989316e+35,9.638578e-07,-2.3670834e+13,4.1168017e+22,2.0537059e-36,-1.863195e+19,-2.974444e+20,1.5068721e-34,-1.4961241e-32,1.0214693e+32,5.245192e-24,1.13598036e-29,-1.6339285e+10,0.00065205223,-1.4312002e+17,9.939042e-36,-1.3663042e+10,1.53824e+10,1.3712913e-33,-1.0928513e-32,-3.6265016e+31,6.461098e-24,-1.954001e-15,2.8298525e+35,5.175261e+14,1.7759765e-27,1.6392416e-16,6.428566e-37,1535.543,-4.1986445e-12,-2.6529006e-17,8.900127e+23,-1.1490072e-06,-0.0017470438,4.0433764e+12,4.654252e+27,4.594365e+20,-2.3189122e+28,-9.51742e+14,-8.018728e-14,-1.1220497e-05,-5463.375,7.6114035e-14,4.2203415e-36,3.8454696e-33,-2.044735e+24,5.038317e+27,-1.5114285e-37,-3.020244e-32,4.708643e-30,-2.8099323e+16,-2.4019435e+16,1.2414438e-25,-4.1356966e+24,4.1722745e+23,5.909729e+30,9.031231e+14,-9.961098e-26,-2.3272292e-15,-8.684913e+07,-9.0688864e+30,0.80774206,-8.4206555e-30,-3.4352218e+08,-7.351497e+16,5.191228e-22,2.0490437e+19,-2.1348477e+31,1.5739561e-23,0.0013782693,-8.223068e-08,-2.889267e-09,2.6164106e-21,0.018983863,6.4987636e+36,-4.6535573e-17,-187115.52,-1.8431035e-15,-3.3648766e-19,-3.993437e+12,-0.117358364,3.474307e+20,-1.6520153e-36,8.7307416e+32,9.214551e+36,4.951576e+11,9.9241624e+11,2.1719176e-21,7.036603e-35,-3.1874566e-07,4155.276,4.9421744e+07,-2.1712303e+14,-3.2527146e-31,0.06682631,-65360.23,-9.724683e+17,1.0676598e+14,52765.777,2.8904676e+26,-1.1165929e+20,1.6973611e-06,1.3360952e-11,-1.2992713e-06,-1.5661425e-34,9.4431965e-21,2.6256044e+25,7.6935e+16,-4.106266e-23,1.287502e-29,-2.8967058e+25,22.140697,1.3532526e-28,-1.5376685e+09,2.08713e-19,-5.170752e+32,-2.0758197e+29,8.51341e+16,-0.49153152,-4.2609165e+29,-2.6895548e-18,2.0870336e+09,-0.116082415,1.344804e-35,-8.4128977e-35,-5.7708005e-13,3.375915e-32,-1.7336768e-38,-2.7936306e-38,1.6190567e+24,-1.6931041e+17,-7.4687314e-06,-2.4972583e-19,-2.6549333e+29,-3.2131524e+31,-2.3428224e+30,7.0493584e+22,2.9461263e-11,-4.5046208e+21,2.3884216e+15,-4.4159202e-14,-5.6446796e+15,715.9431,0.0017357102,-9.917166e-21,-2.6668212e+10,-9.7078955e-26,9.9173184e+08,-2.5764964e-20,-4.307863e+08,1.9696878e-05,-3.5255323e-09,-0.02157946,-3.5310542e-35,-2.0905306e+31,-4.9495177e-32,4.3963204e-34,-8.890641e-15,-1.6562179e-11,3.9860462e+37,-1.7920364e-13,-3.7419047e+31,1.3967947e-35,-5.8442524e-08,1.7042702e-34,-1.84417e+20,1.6697424e+38,1.6462293e-37,1.8198747e-22,-6.3201984e+29,-4.855668e+13,1.2140564e-20,-4.3376446e+12,-6.236808e-09,4.1889122e+06,-8.2706247e+15,-3.5983835e-10,2.8921258e-18,3.079598e-37,-2.8257165e+12,6.102179,2.2939126e-28,2.258509e-37,-6.6644103e+24,5.854955e+36,2.9443493e+11,-8.707032e+35,-2.1845732e-29,-5.2114244e-14,-3.1998103e+18,6.801516e+17,-0.0052741654,-6.281271e+35,2.2779162e-10,-2.934443e-37,-4.1814585e-23,-5.877717e-06,-1.0406358e+29,2.5081794e-17,7.2724314e-12,-1.1049265e+16,1.3718928e+34,-1.3802625e-15,-6.165374e+08,-7.074791e-26,-2.7184587e-05,-2.4021848e+17,3.90424e+15,-8.618735e+18,-1.9121076e-30,8.738285e+36,-1.1199308e+16,-4.5516084e+23,1.5757095e-13,-4.2135078e-20,-6.1127397e-17,-2.7176422e-24,-8.227003e+23,-1.8181847e+34,-6.0303226e-28,3.0740938e+34,3.1412143e+31,-3.3082328e+14,8.62655e-09,3.5104286e-08,4932.9673,-3.447542e-32,-1.6092427e+10,4.5084398e-35,1.3899112e+08,-1.7248166e+13,7.819261e-30,4.1767533e-36,-1.7758362e-16,3.6479432e+14,51069.93,-7.5729973e-28,6.5272874e+29,-3.6364498e+33,-2.8332993e+20,7.02101e-34,-8.741774e-19,6.21202e+10,-4.93776e-36,-1.6176124e+37,1.4326037e-37,8.912696,-5.5011062e+23,-3.9157693e-13,-1.1583432e-20,-1.5835892e-35,7.191149e+15,6.139257e+08,-3.515858e+12,2.2576635e-24,1.0213148e-32,9.1603296e+27,-6.0623445e-30,-2.5932804e+30,3.9221975e-24,-2.5860644e-16,-1.8940701e-06,-7.453531e+30,9.911743e-12,-9.0526886e-10,-245.85033,-1.0137778e+27,1.6963995e+20,1.1210791e+07,9.859639e-32,12.018803,-1.9204342e+11,-9.701502e-34,-1.1447795e+33,-2.3120149e+08,1.037891e+23,-1.5347464e+20,5.873533e-10,2.601372e-18,-3.9467417e-17,-6.793792e-27,-1.3554689e-11,-2.715554e-16,1.2490523e-25,-1.8755744e-22,3.4337168e+11,0.40119994,1.0427442e+12,3.509252e-18,-2.7581017e-09,-0.60790735,8.041574e+33,-6.361483e+28,-6.214633e-26,-3.73486e+34,-3951.4019,1.683737e+16,-2.5887798e+32,9.996037e-06,-0.059069216,-1.4552718e-07,7.2328946e+26,-2.609761e+38,1.5913044e-31,6.9506234e+12,8.796054e-23,-1.8327278e+29,-3.0666195e-11,1.0722527e+27,-0.00021237262,-1.7349926e-36,1.6528809e-09,0.6263685,-1.3906403e-23,3.2513156e-15,6.17748e-20,1.0034097e-21,-0.061291937,-0.00047199466,6.861685e+33,-2.4593631e-17,8.253399e-34,1.5821907e-16,1.5009098e+10,-169003.92,2.4384203e+10,3.6666215e-31,1.4497039e+38,9.023773e-37,-7.5352305e+23,-313.2815,-3.1447348e+13,-7.9839086e+32,3.544249e-07,-1.895284e-36,5.8825975e-13,1.7824192e-24,-1.2929252e+18,1.5096953e+11,-1.335822e-17,-3.8003133e+13,-1.3802191e-34,6.764824e+10,1.1773304e+27,-0.012678721,-2.2547275e+16,4.3547994e+18,1.8272758e+33,-5.819788e+17,-3.282042e-05,-4.5282e-29,2.7338457e+16,1.4091523e+23,2.3924999e+27,-1.0495212e-35,5.5579883e+16,1076.5642,7.3076363,-3.5276215e-27,1.108808e+35,-1.9495124e+32,-4.8715345e+27,8.409832,-1.4692703e+29,2.483913e+17,122397.83,4.106457e-32,8.867036e-38,-3.5349854e-29,2.020276e+34,-2.5378242e-07,-6.8458616e+33,2.5016638e+15,-1.2224521e+26,-2.0183842e-17,-1.7261974e+10,-1.1582705e-08,-3.633052e+15,2.7927238e-06,2.6245981e+14,-844.7549,2.5655025e+32,3.473087e-36,2.8806795e-37,1.0705346e-22,1.258975e+20,2.0336914e+11,6.870751e-24,5.1696156e+18,1105.4308,-6.36811e-08,1.07289195e-20,9.2832575e-05,-1.5246655e+07,1.9494838e-15,7.230135e+08,-4.731663e-20,9.11425e+28,2.1766038e-22,9.881864e-22,-1.8617511e-21,-0.024041481,0.0016246673,6.0144664e+25,-1.2280182e-35,1.7382718e-33,-2.372831e+38,0.020030946,1.05842756e-17,-1.97993e+24,-4.760943e-37,2.0171815e+10,-1.277826e+07,3.738077e+25,1.8690467e+27,9.0490215e+20,-1.3335304e+13,-7.0346785e-25,7.907721e-26,4.4488874e+29,-5.729808e-11,-9.872907e-30,-1.6089304e-36,927.789,-4.9754034e+35,-5.806366e+32,1.5585756e-07,3.819309e-34,-9.583515e-25,5.2101676e+20,5.24073e-26,8.254412e+11,1.06178214e+17,-6.2939953e-22,-0.0058569936,2.1293002e+21,-9.464157e-32,-2.2072352e+06,19.654005,-2.5293177e-16,-4.0277912e+24,-1.1967685e-14,-347176.22,-2.642125e-10,-3.5889257e+11,-3.230497e+35,-8.055738e-21,8.647049e-27,-2.8028652e+13,1.4865494e-31,7.2375544e+35,7.2469724e-27,2.0649533e+27,2.5358454e+28,-1.8076993e-17,5.2007156e-13,-9.720949e+17,251.47394,-4.5960463e-33,3.8532173e-26,7198.818,2.2241141e-33,-8.5995773e+21,-4.8312757e-23,-1.1191778e+36,1.03292986e-22,-7.630812e-20,-9.714141e+31,4.3231003e-07,-6.749065e+13,715713.7,-4.2591575e+35,3.3866454e-21,-9.842541e+14,-3.2271568e-17,8.068436e-10,-8.483994e-32,-3.0685793e+27,4.1277426e+24,4.7438972e-35,-2.5681207e+13,4.9008548e-26,1.2707589e+09,-8.126093e+35,-2.1431158e+20,5.4505226e-19,-61248.53,-6.7359405e+06,1.2406135e+20,-1.9703249e+34,4.4314493e-25,4.4176578e+36,1.7712343e-07,1.6950941e-28,-3.8668055e+22,50719.14,-3.0480184e-35,8.722834e-27,-1.1537289e-23,-1.2731998e+22,-2.2710173e+34,-4.217428e+25,-2.785257e-14,-3.055259e-09,1.8279009e-25,3.6298695e-25,-1.080939e+29,4.2491363e-10,7.572785e+22,-7.8456985e+37,3.9740913e-05,-1.0887203e-23,4.7720243e+35,141927.16,1.20271835e+23,6.63473e-34,1.1906013e-24,7.632316e-30,-2.3508733e-35,-7.58746e-20,7.458457e-14,3.896153e-26,4.725504e+27,1.0725734e-15,1.574169e+06,3.855963e+10,-7.121936e+17,-2.1525178e-13,-2.1420613e+19,7.293989e-18,-1.1784969e-14,4.531935e+36,-3.0131077e+21,5.088653e-20,4.7200237e+28,-5.768743e-28,-1.8867813e+25,-1.152947e+36,0.0015127496,-27.29136,-47128.883,-3.074256e-14,-1.0213085e-32,-1.9518744e+07,-1.2381331e+18,1.3307534e-05,3.0245576e+09,6.786652e-31,-9.489693e+17,1.8264907e-33,2.8924616e-25,5.718142e+31,1.2850974e+08,0.0018981321,-0.9414485,2.983291e-12,-3.2118843e+29,-1.1818716e+12,-8.6862335e-18,5.271352e+30,-9.709545e-15,-5.3912076e+34,-5.795035e-14,2.3083956e+17,-7.406015e-05,-1.5589114e+29,-4.6596954e+33,-6.103788e+26,-5.9471277e+24,4.8696433e+30,0.0012354747,4.675142e+10,-1.4623685e-32,0.33321744,2.8064528e+13,-1.15319815e-36,6.258981e-36,3.6705922e+24,2.7779204e-28,6.141108e+28,-3.1467098e+08,3.9248496e-12,2.2640858e-36,-6.6909075e-29,9.5780116e+30,-0.029649306,3.0790933e+24,1.7250335e-36,1.8408054e-38,-1.096171e-30,-2.172769e+16,-0.21447298,-1.2557303e+15,-1.1547246e+08,6.1212383e-21,1.4873707e-36,4.920212e+09,5.7410375e+12,4.814682e+22,-1.503837e+07,0.002519884,-9.069646e+31,3.5532365e+34,-2.1575285e+23,2564.7178,-2.614844e-37,1.2701972e+38,9.623852e+19,-7.0460447e+27,77.5131,5.148774e+13,-3.6396362e-37,-1.6599959e-05,-3.1524883e-30,-1.7043394e+17,-1.7135373e+20,-1.0626897e+13,-4.068405e+25,1.4437259e+10,-4.049864e-24,3.8314443e+11,-9.8046896e+20,-6.4173795e-34,-1.169193e+20,6.1278707e+21,-2.595911e+29,1.2696151e+20,-1.14963605e-20,1.5447467e-37,-0.43862197,-5.5107807e-24,-1.0578402e-15,-2.4546021e-20,-1.9839618e+34,1.0931126e-08,-5.968573e+27,-5.7065643e-27,-306.16663,-2.280879e+37,-11917.31,6.0676324e+13,-2.7226617e+37,-1.3529594e+16,-8.570867e-18,-8.668089e+09,2.8618707e-37,-1.8226313e-18,1.7294709e+25,-2.2327566e+35,-7.465566e-06,-2.1974653e+21,7.830357e-21,-1927.5461,20596.955,9.477653e-39,2.6598375e-27,-1.2583651e+18,-4.3629502e+32,2.475272e+33,0.0011622943,-4.4202787e+10,-1.4262266e-21,1.0460323e+06,-3877.7295,1.0924367e-35,-0.20075817,2.5885309e+23,-9.690442e+17,2.5387563e+27,9.81975e+06,1.9128665e-05,3.3960005e+36,-5.8138174e-29,-6.079516e+13,3.488598e+32,6.5536805e-21,5.9942668e-06,7.428099e-16,0.006287469,-0.00018362574,-78772.67,-4.960866e-10,1.4816246e-13,-1.0939379e+30,-7.260318e-20,6.4313865e-15,-9.056716e+08,-2.4645713e+23,1.5613149e+07,9.266654e-27,6.436073e+24,-619.01624,-2.0985294e+07,-1.0607246e-25,8.5829706e-05,-5.628521e-33,32278.344,-1.7877486e+08,-7.827001,4310.3013,1.0636503e+19,0.72611856,9.476063e+26,-5.8855694e-23,-43956.297,1.0630598e+37,-1.6215818e+30,-4.1396967e-29,20.5111,5.7685434e+10,2.9919607e+19,9.444301e-30,443.78705,79.07875,-7.568505e+14,-1.0801959e-23,1.0917732e-26,6.3329817e-32,2.0065554e+27,3.0114737e-07,-1.9907938e+07,-4.8319212e+36,-8.940062e+13,-5.9788185e+11,1.167407e+18,9.2426606e-29,-6.319422e+30,-9.68077e+24,-2.1292387e+21,-3.5901788e+06,4.1973945e-07,2.0166974e-19,-20191.2,-1.0056816e-32,1.9466332e+07,1.298115e+33,-1.2232623e-08], col_w0_111_w0_121='2028-09-25 06:24:58.000503', rowid=1045335660121128961 (SQLSTATE 23502)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3228
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:512
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:609
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func4.func6
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:120
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: validation of column "col176_w1_192" NOT NULL failed on row: col47_w0_62='54aa7523-0fb1-4d47-89d1-325341d96435', col47_w0_50='05:41:14.984757', col47_w0_48=!'wVV', col176_w1_184=1266, col176_w1_185='1992-01-21', col176_w1_192=NULL, col176_w1_178=[1.8291579e+25,4.576669e+36,4.027279e+22,0.00012290478,2.76803e-05,1.02607264e+09,1.1482619e+20,2.9291637e+36,-1.832785e-21,605456.94,-7.336126e+34,7.9806176e+12,7.9217315,-7.926753e-38,-4.785446e-15,-4.2700513e+36,-2.1663763e+15,-7.703745e+31,-8.5369656e+09,-0.00011374854,-8.525576e-31,2.196886e-23,1.4807279e-25,1.0925293e+13,-2.1461318e-38,1.5555648e+22,3.0977012e-20,-1.6310505e-35,119088.17,-6.9775663e+25,-1.9890994e+22,-842812.6,1.5902551e+16,-8.4925106e+37,9.262322e-16,-2.146512e-30,1.409276e+37,7.3536126e-37,-4.751217e-27,-6.915994e+13,-3.2004928e+11,7.468967e+13,4.9822395e+33,-6.796619e-10,2.4918778e+08,1.0572669e-18,-11.972408,2.1249514e+20,1.3580067e+31,-4.13354e-39,2.860174e+19,9.7524676e-08,1.281487e-17,5.7726207,-1.9020512e+22,-1.3046642e+22,-1.2268314e+10,2.7600295e-19,22843.742,3.2023303e-07,-844.45245,-2.8693878e+34,-7.3299745e-08,-1.7874349e-34,1.4188787e-36,7.96708e+14,7.305435e+07,1.6642996e-05,7.381059e-31,-7.09088e-07,4.1403404e-29,2.1951314e+09,5.1102694e+08,-4.0989316e+35,9.638578e-07,-2.3670834e+13,4.1168017e+22,2.0537059e-36,-1.863195e+19,-2.974444e+20,1.5068721e-34,-1.4961241e-32,1.0214693e+32,5.245192e-24,1.13598036e-29,-1.6339285e+10,0.00065205223,-1.4312002e+17,9.939042e-36,-1.3663042e+10,1.53824e+10,1.3712913e-33,-1.0928513e-32,-3.6265016e+31,6.461098e-24,-1.954001e-15,2.8298525e+35,5.175261e+14,1.7759765e-27,1.6392416e-16,6.428566e-37,1535.543,-4.1986445e-12,-2.6529006e-17,8.900127e+23,-1.1490072e-06,-0.0017470438,4.0433764e+12,4.654252e+27,4.594365e+20,-2.3189122e+28,-9.51742e+14,-8.018728e-14,-1.1220497e-05,-5463.375,7.6114035e-14,4.2203415e-36,3.8454696e-33,-2.044735e+24,5.038317e+27,-1.5114285e-37,-3.020244e-32,4.708643e-30,-2.8099323e+16,-2.4019435e+16,1.2414438e-25,-4.1356966e+24,4.1722745e+23,5.909729e+30,9.031231e+14,-9.961098e-26,-2.3272292e-15,-8.684913e+07,-9.0688864e+30,0.80774206,-8.4206555e-30,-3.4352218e+08,-7.351497e+16,5.191228e-22,2.0490437e+19,-2.1348477e+31,1.5739561e-23,0.0013782693,-8.223068e-08,-2.889267e-09,2.6164106e-21,0.018983863,6.4987636e+36,-4.6535573e-17,-187115.52,-1.8431035e-15,-3.3648766e-19,-3.993437e+12,-0.117358364,3.474307e+20,-1.6520153e-36,8.7307416e+32,9.214551e+36,4.951576e+11,9.9241624e+11,2.1719176e-21,7.036603e-35,-3.1874566e-07,4155.276,4.9421744e+07,-2.1712303e+14,-3.2527146e-31,0.06682631,-65360.23,-9.724683e+17,1.0676598e+14,52765.777,2.8904676e+26,-1.1165929e+20,1.6973611e-06,1.3360952e-11,-1.2992713e-06,-1.5661425e-34,9.4431965e-21,2.6256044e+25,7.6935e+16,-4.106266e-23,1.287502e-29,-2.8967058e+25,22.140697,1.3532526e-28,-1.5376685e+09,2.08713e-19,-5.170752e+32,-2.0758197e+29,8.51341e+16,-0.49153152,-4.2609165e+29,-2.6895548e-18,2.0870336e+09,-0.116082415,1.344804e-35,-8.4128977e-35,-5.7708005e-13,3.375915e-32,-1.7336768e-38,-2.7936306e-38,1.6190567e+24,-1.6931041e+17,-7.4687314e-06,-2.4972583e-19,-2.6549333e+29,-3.2131524e+31,-2.3428224e+30,7.0493584e+22,2.9461263e-11,-4.5046208e+21,2.3884216e+15,-4.4159202e-14,-5.6446796e+15,715.9431,0.0017357102,-9.917166e-21,-2.6668212e+10,-9.7078955e-26,9.9173184e+08,-2.5764964e-20,-4.307863e+08,1.9696878e-05,-3.5255323e-09,-0.02157946,-3.5310542e-35,-2.0905306e+31,-4.9495177e-32,4.3963204e-34,-8.890641e-15,-1.6562179e-11,3.9860462e+37,-1.7920364e-13,-3.7419047e+31,1.3967947e-35,-5.8442524e-08,1.7042702e-34,-1.84417e+20,1.6697424e+38,1.6462293e-37,1.8198747e-22,-6.3201984e+29,-4.855668e+13,1.2140564e-20,-4.3376446e+12,-6.236808e-09,4.1889122e+06,-8.2706247e+15,-3.5983835e-10,2.8921258e-18,3.079598e-37,-2.8257165e+12,6.102179,2.2939126e-28,2.258509e-37,-6.6644103e+24,5.854955e+36,2.9443493e+11,-8.707032e+35,-2.1845732e-29,-5.2114244e-14,-3.1998103e+18,6.801516e+17,-0.0052741654,-6.281271e+35,2.2779162e-10,-2.934443e-37,-4.1814585e-23,-5.877717e-06,-1.0406358e+29,2.5081794e-17,7.2724314e-12,-1.1049265e+16,1.3718928e+34,-1.3802625e-15,-6.165374e+08,-7.074791e-26,-2.7184587e-05,-2.4021848e+17,3.90424e+15,-8.618735e+18,-1.9121076e-30,8.738285e+36,-1.1199308e+16,-4.5516084e+23,1.5757095e-13,-4.2135078e-20,-6.1127397e-17,-2.7176422e-24,-8.227003e+23,-1.8181847e+34,-6.0303226e-28,3.0740938e+34,3.1412143e+31,-3.3082328e+14,8.62655e-09,3.5104286e-08,4932.9673,-3.447542e-32,-1.6092427e+10,4.5084398e-35,1.3899112e+08,-1.7248166e+13,7.819261e-30,4.1767533e-36,-1.7758362e-16,3.6479432e+14,51069.93,-7.5729973e-28,6.5272874e+29,-3.6364498e+33,-2.8332993e+20,7.02101e-34,-8.741774e-19,6.21202e+10,-4.93776e-36,-1.6176124e+37,1.4326037e-37,8.912696,-5.5011062e+23,-3.9157693e-13,-1.1583432e-20,-1.5835892e-35,7.191149e+15,6.139257e+08,-3.515858e+12,2.2576635e-24,1.0213148e-32,9.1603296e+27,-6.0623445e-30,-2.5932804e+30,3.9221975e-24,-2.5860644e-16,-1.8940701e-06,-7.453531e+30,9.911743e-12,-9.0526886e-10,-245.85033,-1.0137778e+27,1.6963995e+20,1.1210791e+07,9.859639e-32,12.018803,-1.9204342e+11,-9.701502e-34,-1.1447795e+33,-2.3120149e+08,1.037891e+23,-1.5347464e+20,5.873533e-10,2.601372e-18,-3.9467417e-17,-6.793792e-27,-1.3554689e-11,-2.715554e-16,1.2490523e-25,-1.8755744e-22,3.4337168e+11,0.40119994,1.0427442e+12,3.509252e-18,-2.7581017e-09,-0.60790735,8.041574e+33,-6.361483e+28,-6.214633e-26,-3.73486e+34,-3951.4019,1.683737e+16,-2.5887798e+32,9.996037e-06,-0.059069216,-1.4552718e-07,7.2328946e+26,-2.609761e+38,1.5913044e-31,6.9506234e+12,8.796054e-23,-1.8327278e+29,-3.0666195e-11,1.0722527e+27,-0.00021237262,-1.7349926e-36,1.6528809e-09,0.6263685,-1.3906403e-23,3.2513156e-15,6.17748e-20,1.0034097e-21,-0.061291937,-0.00047199466,6.861685e+33,-2.4593631e-17,8.253399e-34,1.5821907e-16,1.5009098e+10,-169003.92,2.4384203e+10,3.6666215e-31,1.4497039e+38,9.023773e-37,-7.5352305e+23,-313.2815,-3.1447348e+13,-7.9839086e+32,3.544249e-07,-1.895284e-36,5.8825975e-13,1.7824192e-24,-1.2929252e+18,1.5096953e+11,-1.335822e-17,-3.8003133e+13,-1.3802191e-34,6.764824e+10,1.1773304e+27,-0.012678721,-2.2547275e+16,4.3547994e+18,1.8272758e+33,-5.819788e+17,-3.282042e-05,-4.5282e-29,2.7338457e+16,1.4091523e+23,2.3924999e+27,-1.0495212e-35,5.5579883e+16,1076.5642,7.3076363,-3.5276215e-27,1.108808e+35,-1.9495124e+32,-4.8715345e+27,8.409832,-1.4692703e+29,2.483913e+17,122397.83,4.106457e-32,8.867036e-38,-3.5349854e-29,2.020276e+34,-2.5378242e-07,-6.8458616e+33,2.5016638e+15,-1.2224521e+26,-2.0183842e-17,-1.7261974e+10,-1.1582705e-08,-3.633052e+15,2.7927238e-06,2.6245981e+14,-844.7549,2.5655025e+32,3.473087e-36,2.8806795e-37,1.0705346e-22,1.258975e+20,2.0336914e+11,6.870751e-24,5.1696156e+18,1105.4308,-6.36811e-08,1.07289195e-20,9.2832575e-05,-1.5246655e+07,1.9494838e-15,7.230135e+08,-4.731663e-20,9.11425e+28,2.1766038e-22,9.881864e-22,-1.8617511e-21,-0.024041481,0.0016246673,6.0144664e+25,-1.2280182e-35,1.7382718e-33,-2.372831e+38,0.020030946,1.05842756e-17,-1.97993e+24,-4.760943e-37,2.0171815e+10,-1.277826e+07,3.738077e+25,1.8690467e+27,9.0490215e+20,-1.3335304e+13,-7.0346785e-25,7.907721e-26,4.4488874e+29,-5.729808e-11,-9.872907e-30,-1.6089304e-36,927.789,-4.9754034e+35,-5.806366e+32,1.5585756e-07,3.819309e-34,-9.583515e-25,5.2101676e+20,5.24073e-26,8.254412e+11,1.06178214e+17,-6.2939953e-22,-0.0058569936,2.1293002e+21,-9.464157e-32,-2.2072352e+06,19.654005,-2.5293177e-16,-4.0277912e+24,-1.1967685e-14,-347176.22,-2.642125e-10,-3.5889257e+11,-3.230497e+35,-8.055738e-21,8.647049e-27,-2.8028652e+13,1.4865494e-31,7.2375544e+35,7.2469724e-27,2.0649533e+27,2.5358454e+28,-1.8076993e-17,5.2007156e-13,-9.720949e+17,251.47394,-4.5960463e-33,3.8532173e-26,7198.818,2.2241141e-33,-8.5995773e+21,-4.8312757e-23,-1.1191778e+36,1.03292986e-22,-7.630812e-20,-9.714141e+31,4.3231003e-07,-6.749065e+13,715713.7,-4.2591575e+35,3.3866454e-21,-9.842541e+14,-3.2271568e-17,8.068436e-10,-8.483994e-32,-3.0685793e+27,4.1277426e+24,4.7438972e-35,-2.5681207e+13,4.9008548e-26,1.2707589e+09,-8.126093e+35,-2.1431158e+20,5.4505226e-19,-61248.53,-6.7359405e+06,1.2406135e+20,-1.9703249e+34,4.4314493e-25,4.4176578e+36,1.7712343e-07,1.6950941e-28,-3.8668055e+22,50719.14,-3.0480184e-35,8.722834e-27,-1.1537289e-23,-1.2731998e+22,-2.2710173e+34,-4.217428e+25,-2.785257e-14,-3.055259e-09,1.8279009e-25,3.6298695e-25,-1.080939e+29,4.2491363e-10,7.572785e+22,-7.8456985e+37,3.9740913e-05,-1.0887203e-23,4.7720243e+35,141927.16,1.20271835e+23,6.63473e-34,1.1906013e-24,7.632316e-30,-2.3508733e-35,-7.58746e-20,7.458457e-14,3.896153e-26,4.725504e+27,1.0725734e-15,1.574169e+06,3.855963e+10,-7.121936e+17,-2.1525178e-13,-2.1420613e+19,7.293989e-18,-1.1784969e-14,4.531935e+36,-3.0131077e+21,5.088653e-20,4.7200237e+28,-5.768743e-28,-1.8867813e+25,-1.152947e+36,0.0015127496,-27.29136,-47128.883,-3.074256e-14,-1.0213085e-32,-1.9518744e+07,-1.2381331e+18,1.3307534e-05,3.0245576e+09,6.786652e-31,-9.489693e+17,1.8264907e-33,2.8924616e-25,5.718142e+31,1.2850974e+08,0.0018981321,-0.9414485,2.983291e-12,-3.2118843e+29,-1.1818716e+12,-8.6862335e-18,5.271352e+30,-9.709545e-15,-5.3912076e+34,-5.795035e-14,2.3083956e+17,-7.406015e-05,-1.5589114e+29,-4.6596954e+33,-6.103788e+26,-5.9471277e+24,4.8696433e+30,0.0012354747,4.675142e+10,-1.4623685e-32,0.33321744,2.8064528e+13,-1.15319815e-36,6.258981e-36,3.6705922e+24,2.7779204e-28,6.141108e+28,-3.1467098e+08,3.9248496e-12,2.2640858e-36,-6.6909075e-29,9.5780116e+30,-0.029649306,3.0790933e+24,1.7250335e-36,1.8408054e-38,-1.096171e-30,-2.172769e+16,-0.21447298,-1.2557303e+15,-1.1547246e+08,6.1212383e-21,1.4873707e-36,4.920212e+09,5.7410375e+12,4.814682e+22,-1.503837e+07,0.002519884,-9.069646e+31,3.5532365e+34,-2.1575285e+23,2564.7178,-2.614844e-37,1.2701972e+38,9.623852e+19,-7.0460447e+27,77.5131,5.148774e+13,-3.6396362e-37,-1.6599959e-05,-3.1524883e-30,-1.7043394e+17,-1.7135373e+20,-1.0626897e+13,-4.068405e+25,1.4437259e+10,-4.049864e-24,3.8314443e+11,-9.8046896e+20,-6.4173795e-34,-1.169193e+20,6.1278707e+21,-2.595911e+29,1.2696151e+20,-1.14963605e-20,1.5447467e-37,-0.43862197,-5.5107807e-24,-1.0578402e-15,-2.4546021e-20,-1.9839618e+34,1.0931126e-08,-5.968573e+27,-5.7065643e-27,-306.16663,-2.280879e+37,-11917.31,6.0676324e+13,-2.7226617e+37,-1.3529594e+16,-8.570867e-18,-8.668089e+09,2.8618707e-37,-1.8226313e-18,1.7294709e+25,-2.2327566e+35,-7.465566e-06,-2.1974653e+21,7.830357e-21,-1927.5461,20596.955,9.477653e-39,2.6598375e-27,-1.2583651e+18,-4.3629502e+32,2.475272e+33,0.0011622943,-4.4202787e+10,-1.4262266e-21,1.0460323e+06,-3877.7295,1.0924367e-35,-0.20075817,2.5885309e+23,-9.690442e+17,2.5387563e+27,9.81975e+06,1.9128665e-05,3.3960005e+36,-5.8138174e-29,-6.079516e+13,3.488598e+32,6.5536805e-21,5.9942668e-06,7.428099e-16,0.006287469,-0.00018362574,-78772.67,-4.960866e-10,1.4816246e-13,-1.0939379e+30,-7.260318e-20,6.4313865e-15,-9.056716e+08,-2.4645713e+23,1.5613149e+07,9.266654e-27,6.436073e+24,-619.01624,-2.0985294e+07,-1.0607246e-25,8.5829706e-05,-5.628521e-33,32278.344,-1.7877486e+08,-7.827001,4310.3013,1.0636503e+19,0.72611856,9.476063e+26,-5.8855694e-23,-43956.297,1.0630598e+37,-1.6215818e+30,-4.1396967e-29,20.5111,5.7685434e+10,2.9919607e+19,9.444301e-30,443.78705,79.07875,-7.568505e+14,-1.0801959e-23,1.0917732e-26,6.3329817e-32,2.0065554e+27,3.0114737e-07,-1.9907938e+07,-4.8319212e+36,-8.940062e+13,-5.9788185e+11,1.167407e+18,9.2426606e-29,-6.319422e+30,-9.68077e+24,-2.1292387e+21,-3.5901788e+06,4.1973945e-07,2.0166974e-19,-20191.2,-1.0056816e-32,1.9466332e+07,1.298115e+33,-1.2232623e-08], col_w0_111_w0_121='2028-09-25 06:24:58.000503', rowid=1045335660121128961 (SQLSTATE 23502)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 156
Total Schema Statements Succeeded = 156
Total Schema Statement Expected Failures = 1
Total Transactions Committed = 101
Total Transactions Rolled Back = 79
Total Transactions Executed = 180
    schema_change_external_test.go:96: backup, tracing data, and system table dumps in /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_5_of_25/test.outputs/logTestWorkload1508928707
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload1508928707
--- FAIL: TestWorkload (20.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #140702 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-24.3.5-rc release-blocker]
- #140384 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [duplicate constraint name] [C-test-failure O-robot P-3 T-sql-foundations branch-release-24.1.12-rc]
- #139864 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.1]
- #136966 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [duplicate key value violates unique constraint] [C-test-failure O-robot P-2 T-sql-foundations branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -405,6 +405,9 @@ func (og *operationGenerator) addUniqueConstraint(ctx context.Context, tx pgx.Tx
 
 	if !canApplyConstraint {
 		og.candidateExpectedCommitErrors.add(pgcode.UniqueViolation)
+		// For newly created tables its possible for the execution to
+		// the unique violation error.
+		stmt.potentialExecErrors.add(pgcode.UniqueViolation)
 	} else {
 		// Otherwise there is still a possibility for an error,
 		// so add it in the potential set, since our validation query
@@ -1157,6 +1160,11 @@ func (og *operationGenerator) createIndex(ctx context.Context, tx pgx.Tx) (*opSt
 			{code: pgcode.FeatureNotSupported, condition: hasAlterPKSchemaChange && !og.useDeclarativeSchemaChanger},
 			{code: pgcode.FeatureNotSupported, condition: lastColInvertedIndexIsDescending},
 			{code: pgcode.FeatureNotSupported, condition: pkColUsedInInvertedIndex},
+		})
+		// Unique violations can occur at the statement phase if the table is
+		// new.
+		stmt.potentialExecErrors.addAll(codesWithConditions{
+			{code: pgcode.UniqueViolation, condition: !uniqueViolationWillNotOccur},
 		})
 	}
 
@@ -2476,6 +2484,9 @@ func (og *operationGenerator) setColumnNotNull(ctx context.Context, tx pgx.Tx) (
 		}
 		if colContainsNull {
 			og.candidateExpectedCommitErrors.add(pgcode.NotNullViolation)
+			// If the table is created within the txn, then the not null violation
+			// will be an execution error.
+			stmt.potentialExecErrors.add(pgcode.NotNullViolation)
 		}
 		// If we are running with the legacy schema changer, the not null constraint
 		// is enforced during the job phase. So it's still possible to INSERT not null

